src/js/media-chrome-dialog.ts

```diff
@@ -94,19 +96,26 @@ class MediaChromeDialog extends globalThis.HTMLElement {
   constructor() {
     super();
 
+    this.addEventListener('invoke', this);
+    this.addEventListener('focusout', this);
+    this.addEventListener('keydown', this);
+  }
+
+  #init() {
+    if (this.#isInit) return;
+    this.#isInit = true;
+
     if (!this.shadowRoot) {
       // Set up the Shadow DOM if not using Declarative Shadow DOM.
       this.attachShadow({ mode: 'open' });
 
-      this.nativeEl = (
-        this.constructor as typeof MediaChromeDialog
-      ).template.content.cloneNode(true) as HTMLElement;
-      this.shadowRoot.append(this.nativeEl);
-    }
+      const attrs = namedNodeMapToObject(this.attributes);
 
-    this.addEventListener('invoke', this);
-    this.addEventListener('focusout', this);
-    this.addEventListener('keydown', this);
+      this.shadowRoot.innerHTML = /*html*/ `
+        ${(this.constructor as typeof MediaChromeDialog).getTemplateHTML(attrs)}
```

Comment:
question: assuming we keep something roughly resembling this architecture, why use attributes instead of the type-accurate properties for these functions?

Reply:
this will also be used to SSR so it should output plain HTML

Reply:
Cool that def makes sense with the additional arguments. Sold.